### PR TITLE
e2e: check topology for errors, fix hotpluggable memory qemu opts

### PR DESCRIPTION
**Pull request by askervin**

### CODE REVIEW

File: demo/lib/topology2qemuopts.py [hunk 1]
```diff
@@ -148,6 +174,8 @@ def dists(numalist):
         if "node-dist" in numaspec:
             for n in range(first_node_in_group, lastnode_in_group):
                 node_node_dist[n] = {int(nodename): value for nodename, value in numaspec["node-dist"].items()}
+    if lastnode_in_group == -1:
```
Comment (marquiz) [inline demo/lib/topology2qemuopts.py:177]: nitnit: would it be safer/saner to do `< 0`

File: demo/lib/topology2qemuopts.py [hunk 2]
```diff
@@ -197,10 +226,23 @@ def qemuopts(numalist):
...
 
     # Read cpu counts, and "mem" and "nvmem" sizes for all nodes.
+    threadcount = -1
     for numalistindex, numaspec in enumerate(numalist):
         nodecount = int(numaspec.get("nodes", 1))
         groupnodes[numalistindex] = tuple(range(lastnode + 1, lastnode + 1 + nodecount))
-        threadcount = int(numaspec.get("threads", 2)) # threads per cpu
+        corecount = int(numaspec.get("cores", 0))
+        if corecount > 0:
+            if threadcount == -1:
```
Comment (marquiz) [inline demo/lib/topology2qemuopts.py:235]: nitnit :I'd probably use `< 0` here, too

File: demo/lib/topology2qemuopts.py [hunk 3]
```diff
@@ -197,10 +226,23 @@ def qemuopts(numalist):
...
+                # threads per cpu, set only once based on the first cpu-ful numa node
+                threadcount = int(numaspec.get("threads", 2))
+                threads_set_node = numaspec
+            else:
+                # threadcount already set, only check that there is no mismatch
+                if (not numaspec.get("threads", None) is None and
```
Comment (marquiz) [inline demo/lib/topology2qemuopts.py:241]: nit: I think `is not None` would be more python-esque and possibly easier to read(?)

**Review by marquiz — COMMENTED** [review]
I really do not have anything to say about the functionality, just few tiny nits on style. So, I'd say LGTM from my side

**Review by klihub — COMMENTED** [review]
Can't comment much on the functional changes.

File: demo/lib/topology2qemuopts.py (hunk 1 above)
Comment (askervin) [reply demo/lib/topology2qemuopts.py]: Thanks, fixed!

File: demo/lib/topology2qemuopts.py (hunk 2 above)
Comment (askervin) [reply demo/lib/topology2qemuopts.py]: Thanks, fixed!

File: demo/lib/topology2qemuopts.py (hunk 3 above)
Comment (askervin) [reply demo/lib/topology2qemuopts.py]: Thanks, fixed! And changed other `not X is None`'s too.

**Review by marquiz — APPROVED** [review]
I trust on @askervin on the functionality 😉  Code looks good/clean
LGTM

**Pull request merged**